components/ai_chat/renderer/page_content_extractor_browsertest.cc

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/ai_chat/renderer/page_content_extractor.h"
+
+#include <memory>
+#include <optional>
+#include <string>
+
+#include "base/test/test_future.h"
+#include "base/values.h"
+#include "brave/components/ai_chat/core/common/mojom/page_content_extractor.mojom.h"
+#include "content/public/renderer/render_frame.h"
+#include "content/public/test/render_view_test.h"
+#include "services/service_manager/public/cpp/binder_registry.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "third_party/blink/public/web/web_document.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+#include "third_party/blink/public/web/web_script_source.h"
+
+namespace ai_chat {
+
+class PageContentExtractorRenderViewTest : public content::RenderViewTest {
+ public:
+  PageContentExtractorRenderViewTest() = default;
+  ~PageContentExtractorRenderViewTest() override = default;
+
+  void SetUp() override {
+    content::RenderViewTest::SetUp();
+    registry_ = std::make_unique<service_manager::BinderRegistry>();
+  }
+
+  void TearDown() override {
+    registry_.reset();
+    content::RenderViewTest::TearDown();
+  }
+
+  // Helper method to simulate loading a page with a specific URL
+  void LoadPageWithUrl(const std::string& url) {
+    LoadHTMLWithUrlOverride("<html><body>Test page</body></html>", url);
+  }
+
+  // Helper method to execute JavaScript and get the result
+  std::optional<base::Value> ExecuteScript(const std::string& script) {
+    blink::WebScriptSource source(blink::WebString::FromUTF8(script));
+    base::test::TestFuture<std::optional<base::Value>, base::TimeTicks> future;
+
+    GetMainRenderFrame()->GetWebFrame()->RequestExecuteScript(
+        0, base::span_from_ref(source),
+        blink::mojom::UserActivationOption::kDoNotActivate,
+        blink::mojom::EvaluationTiming::kSynchronous,
+        blink::mojom::LoadEventBlockingOption::kDoNotBlock,
+        future.GetCallback(), blink::BackForwardCacheAware::kAllow,
+        blink::mojom::WantResultOption::kWantResult,
+        blink::mojom::PromiseResultOption::kAwait);
+
+    auto [result, start_time] = future.Take();
+    return std::move(result);
+  }
+
+  // Helper method to test ExtractPageContent through mojom interface
+  mojom::PageContentPtr ExtractPageContent() {
+    // Create the extractor instance to register the interface
+    auto extractor = std::make_unique<PageContentExtractor>(
+        GetMainRenderFrame(), registry_.get(), 0, 1);
+    if (!extractor) {
```

Comment:
how could this ever be null?

Reply:
fixed in cabb6bf6adec75e8eff5c756e69a9ce000aa1bd5